rules/coding-style/src/Rector/Variable/UnderscoreToCamelCaseVariableNameRector.php

```diff
@@ -87,7 +96,53 @@ public function refactor(Node $node): ?Node
         }
 
         $node->name = $camelCaseName;
+        $this->updateDocblock($node, $nodeName, $camelCaseName);
 
         return $node;
     }
+
+    private function updateDocblock(Variable $variable, string $variableName, string $camelCaseName): void
+    {
+        $parentNode = $variable->getAttribute(AttributeKey::PARENT_NODE);
+        while ($parentNode) {
+            /** @var ClassMethod|Function_ $parentNode */
+            $parentNode = $parentNode->getAttribute(AttributeKey::PARENT_NODE);
+            if ($parentNode instanceof ClassMethod || $parentNode instanceof Function_) {
+                break;
+            }
+        }
+
+        if ($parentNode === null) {
+            return;
+        }
+
+        $docComment = $parentNode->getDocComment();
+        if ($docComment === null) {
+            return;
+        }
+
+        if ($docComment->getText() === null) {
+            return;
+        }
+
+        if (! Strings::match($docComment->getText(), sprintf(self::PARAM_NAME_REGEX, $variableName))) {
+            return;
+        }
+
+        $newdocComment = Strings::replace(
+            $docComment->getText(),
+            sprintf(self::PARAM_NAME_REGEX, $variableName),
+            function ($match) use ($camelCaseName): string {
+                $match['paramName'] = $camelCaseName;
+                return $match['paramPrefix'] . $match['paramName'];
+            }
+        );
+
+        if ($parentNode instanceof ClassMethod) {
+            $variable->getAttribute(AttributeKey::METHOD_NODE)->setDocComment(new Doc($newdocComment));
```

Comment:
I tried to implement upate `@param` when updating to camel case, but this is not works: 

```bash
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 {
        /**
      * @param string $x
-     * @param string $yZ
+     * @param string $y_z
      */
     public function run(string $x, string $yZ)
```

I printed the `$newdocComment` and it shows printed correctly: 

```bash
....^ """
/**\n
     * @param string $x\n
     * @param string $yZ\n
     */
"""
```

I tried both `$parentNode` and use `$variable` to get method node, both not works. @TomasVotruba do you have some hints for it? Thank you.

Reply:
There is a rule that change param name in docblock too:
https://github.com/rectorphp/rector/blob/master/rules/naming/tests/Rector/ClassMethod/RenameParamToMatchTypeRector/Fixture/update_doc_name.php.inc

This might help:
https://github.com/rectorphp/rector/blob/219069f29f2d8e4ed1fe954d3747ea86b2a153cf/rules/naming/src/Rector/ClassMethod/RenameParamToMatchTypeRector.php#L122-L123

Reply:
It works, thank you, it is implemented.